spec/syntax.md

```diff
@@ -309,6 +309,8 @@ A _well-formed_ message is considered _valid_ if the following requirements are
 
 - The number of keys on each _variant_ MUST be equal to the number of _selectors_.
 - At least one _variant's_ keys MUST all be equal to the catch-all key (`*`).
+- Each _selector_ MUST have an _annotation_,
+  or contain a _variable_ that references a _declaration_ with an _annotation_.
```

Comment:
Would this make the following message invalid?

```
let $foo = {$count :plural}
let $bar = {$foo}
let $baz = {$bar}
match {$baz} ...
```

Reply:
Yes.

Reply:
Ah, I see your "and that's just silly" comment in the PR's description. While I agree that it's not something I'd expect to see in a real-life message, I think it's an important litmus test for how easy it is to build a mental model of how a message is evaluated.

Adding arbitrary rules to this effect ("references can have as many jumps as needed, unless used in a selector, when it can only be one jump to an expression with an annotation") is going to make MF2 harder to learn.

Furthermore, such rules don't really help, because as a translator, I can still do to bypass such checks:

```
let $foo = {$count :noop}
match {$foo} ...
```

Nor do they prevent us from errors when non-matching functions are used:

```
let $foo = {$count :datetime}
match {$foo} ...
```

Reply:
```suggestion
  or contain a _variable_ that directly or indirectly references a _declaration_ with an _annotation_.
```